### PR TITLE
MM-16174  Migrate "Team.SaveMember" to Sync by default #11125

### LINKED ISSUE #11125: Migrate "Team.SaveMember" to Sync by default
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

----

**Notes**: [Jira ticket](https://mattermost.atlassian.net/browse/MM-16174)

Mattermost is migrating its Store layer to be sync by default, and only use Async when needed and we're looking for contributors to help with that effort. This Help Wanted issue is to migrate the `SaveMember` in the `Team` store.

The expected way to implement it is, go to the `Team` store implementation in the `store/sqlstore/` directory, modify the method `SaveMember` to return directly an object from the `model` module, and a `*model.AppError` (removing the `store.Do` wrapper). After that, you must modify the interface defined in `store/store.go` to match with the changes. Then, you should execute `make store-mocks` to rebuild the mocks with the new interface. And finally, modify the rest of the code (tests included) to use the new interface of the function properly.

Example: mattermost/mattermost-server#10613

### PR DESCRIPTION
**Pull request by Inconnu08**
#### Summary
Migrate `Team.SaveMember` to sync by default, removing the `return.Do` wrapper.

#### Ticket Link
Fixes #11125

### CODE REVIEW

File: app/auto_users.go [hunk 1]
```diff
@@ -48,6 +48,11 @@ func (a *App) CreateBasicUser(client *model.Client4) *model.AppError {
 		if resp.Error != nil {
 			return resp.Error
 		}
+		store.Must(a.Srv.Store.User().VerifyEmail(ruser.Id, ruser.Email))
```
Comment (jespino) [inline app/auto_users.go]: Why are this changes added here?

File: store/sqlstore/team_store.go [hunk 2]
```diff
@@ -491,18 +491,17 @@ func (s SqlTeamStore) getTeamMembersWithSchemeSelectQuery() sq.SelectBuilder {
...
-		dbMember := NewTeamMemberFromModel(member)
+	dbMember := NewTeamMemberFromModel(member)
 
-		if maxUsersPerTeam >= 0 {
-			count, err := s.GetMaster().SelectInt(
-				`SELECT
+	if maxUsersPerTeam >= 0 {
+		count, err := s.GetMaster().SelectInt(
+			`SELECT
 					COUNT(0)
```
Comment (jespino) [inline store/sqlstore/team_store.go]: do you mind to indent accordingly the rest of the query text (1 tab less)?

File: store/storetest/compliance_store.go [hunk 3]
```diff
@@ -360,20 +366,21 @@ func testMessageExportPrivateChannel(t *testing.T, ss store.Store) {
 		Username: model.NewId(),
 	}
 	user1 = store.Must(ss.User().Save(user1)).(*model.User)
-	store.Must(ss.Team().SaveMember(&model.TeamMember{
+	_, err = ss.Team().SaveMember(&model.TeamMember{
 		TeamId: team.Id,
 		UserId: user1.Id,
-	}, -1))
+	}, -1)
```
Comment (jespino) [inline store/storetest/compliance_store.go]: missing the `require.Nil(t, err)` here

File: store/storetest/compliance_store.go [hunk 4]
```diff
@@ -360,20 +366,21 @@ func testMessageExportPrivateChannel(t *testing.T, ss store.Store) {
...
 		Email:    MakeEmail(),
 		Username: model.NewId(),
 	}
 	user2 = store.Must(ss.User().Save(user2)).(*model.User)
-	store.Must(ss.Team().SaveMember(&model.TeamMember{
+	_, err = ss.Team().SaveMember(&model.TeamMember{
 		TeamId: team.Id,
 		UserId: user2.Id,
-	}, -1))
+	}, -1)
```
Comment (jespino) [inline store/storetest/compliance_store.go]: and here

File: store/storetest/team_store.go [hunk 5]
```diff
@@ -345,7 +345,7 @@ func testTeamStoreSearchPrivate(t *testing.T, ss store.Store) {
 	testCases := []struct {
 		Name            string
 		Term            string
-		ExpectedLenth   int
+		ExpectedLength  int
```
Comment (jespino) [inline store/storetest/team_store.go]: nice catch :heart:

File: store/storetest/team_store.go [hunk 6]
```diff
@@ -1433,9 +1453,9 @@ func testTeamStoreMigrateTeamMembers(t *testing.T, ss store.Store) {
 		ExplicitRoles: "something_else",
 	}
 
-	tm1 = (<-ss.Team().SaveMember(tm1, -1)).Data.(*model.TeamMember)
-	tm2 = (<-ss.Team().SaveMember(tm2, -1)).Data.(*model.TeamMember)
-	tm3 = (<-ss.Team().SaveMember(tm3, -1)).Data.(*model.TeamMember)
+	tm1, _ = ss.Team().SaveMember(tm1, -1)
```
Comment (jespino) [inline store/storetest/team_store.go]: not in the previous code, but we can add the `require.Nil(t, err)` here too (and in the next 2 lines)

File: store/storetest/user_store.go [hunk 7]
```diff
@@ -212,7 +217,7 @@ func testUserStoreUpdateUpdateAt(t *testing.T, ss store.Store) {
 	u1.Email = MakeEmail()
 	store.Must(ss.User().Save(u1))
 	defer func() { require.Nil(t, ss.User().PermanentDelete(u1.Id)) }()
-	store.Must(ss.Team().SaveMember(&model.TeamMember{TeamId: model.NewId(), UserId: u1.Id}, -1))
+	_, err := ss.Team().SaveMember(&model.TeamMember{TeamId: model.NewId(), UserId: u1.Id}, -1)
```
Comment (jespino) [inline store/storetest/user_store.go]: I miss the `require.Nil(t, err)` here

File: store/storetest/user_store.go [hunk 8]
```diff
@@ -543,23 +555,27 @@ func testUserStoreGetProfiles(t *testing.T, ss store.Store) {
 	u3.IsBot = true
 	defer func() { store.Must(ss.Bot().PermanentDelete(u3.Id)) }()
 	defer func() { require.Nil(t, ss.User().PermanentDelete(u3.Id)) }()
-	store.Must(ss.Team().SaveMember(&model.TeamMember{TeamId: teamId, UserId: u3.Id}, -1))
+	_, err = ss.Team().SaveMember(&model.TeamMember{TeamId: teamId, UserId: u3.Id}, -1)
+	require.Nil(t, err)
+	require.Nil(t, err)
```
Comment (jespino) [inline store/storetest/user_store.go]: duplicated line

File: store/storetest/user_store.go [hunk 9]
```diff
@@ -2020,11 +2085,11 @@ func testUserStoreSearch(t *testing.T, ss store.Store) {
 	defer func() { require.Nil(t, ss.User().PermanentDelete(u6.Id)) }()
 
 	tid := model.NewId()
-	store.Must(ss.Team().SaveMember(&model.TeamMember{TeamId: tid, UserId: u1.Id}, -1))
-	store.Must(ss.Team().SaveMember(&model.TeamMember{TeamId: tid, UserId: u2.Id}, -1))
-	store.Must(ss.Team().SaveMember(&model.TeamMember{TeamId: tid, UserId: u3.Id}, -1))
-	store.Must(ss.Team().SaveMember(&model.TeamMember{TeamId: tid, UserId: u5.Id}, -1))
-	store.Must(ss.Team().SaveMember(&model.TeamMember{TeamId: tid, UserId: u6.Id}, -1))
+	_, _ = ss.Team().SaveMember(&model.TeamMember{TeamId: tid, UserId: u1.Id}, -1)
```
Comment (jespino) [inline store/storetest/user_store.go]: I miss the `require.Nil(t, err)` here and in the next 4 lines

File: store/storetest/user_store.go [hunk 10]
```diff
@@ -2337,9 +2402,9 @@ func testUserStoreSearchNotInChannel(t *testing.T, ss store.Store) {
 	defer func() { store.Must(ss.Bot().PermanentDelete(u3.Id)) }()
 
 	tid := model.NewId()
-	store.Must(ss.Team().SaveMember(&model.TeamMember{TeamId: tid, UserId: u1.Id}, -1))
-	store.Must(ss.Team().SaveMember(&model.TeamMember{TeamId: tid, UserId: u2.Id}, -1))
-	store.Must(ss.Team().SaveMember(&model.TeamMember{TeamId: tid, UserId: u3.Id}, -1))
+	_, _ = ss.Team().SaveMember(&model.TeamMember{TeamId: tid, UserId: u1.Id}, -1)
```
Comment (jespino) [inline store/storetest/user_store.go]: I miss the require.Nil(t, err) here and in the next 2 lines

File: store/storetest/user_store.go [hunk 11]
```diff
@@ -2553,9 +2618,9 @@ func testUserStoreSearchInChannel(t *testing.T, ss store.Store) {
...
```
Comment (jespino) [inline store/storetest/user_store.go]: I miss the `require.Nil(t, err)` here and in the next 2 lines

File: store/storetest/user_store.go [hunk 12]
```diff
@@ -2736,15 +2801,15 @@ func testUserStoreSearchNotInTeam(t *testing.T, ss store.Store) {
 	defer func() { require.Nil(t, ss.User().PermanentDelete(u6.Id)) }()
 
 	teamId1 := model.NewId()
-	store.Must(ss.Team().SaveMember(&model.TeamMember{TeamId: teamId1, UserId: u1.Id}, -1))
-	store.Must(ss.Team().SaveMember(&model.TeamMember{TeamId: teamId1, UserId: u2.Id}, -1))
-	store.Must(ss.Team().SaveMember(&model.TeamMember{TeamId: teamId1, UserId: u3.Id}, -1))
+	_, _ = ss.Team().SaveMember(&model.TeamMember{TeamId: teamId1, UserId: u1.Id}, -1)
```
Comment (jespino) [inline store/storetest/user_store.go]: I miss the `require.Nil(t, err)` here and in the next 2 lines

File: store/storetest/user_store.go [hunk 13]
```diff
@@ -2736,15 +2801,15 @@ func testUserStoreSearchNotInTeam(t *testing.T, ss store.Store) {
...
+	_, _ = ss.Team().SaveMember(&model.TeamMember{TeamId: teamId1, UserId: u2.Id}, -1)
+	_, _ = ss.Team().SaveMember(&model.TeamMember{TeamId: teamId1, UserId: u3.Id}, -1)
 	// u4 is not in team 1
-	store.Must(ss.Team().SaveMember(&model.TeamMember{TeamId: teamId1, UserId: u5.Id}, -1))
-	store.Must(ss.Team().SaveMember(&model.TeamMember{TeamId: teamId1, UserId: u6.Id}, -1))
+	_, _ = ss.Team().SaveMember(&model.TeamMember{TeamId: teamId1, UserId: u5.Id}, -1)
```
Comment (jespino) [inline store/storetest/user_store.go]: I miss the `require.Nil(t, err)` here and in the next line

File: store/storetest/user_store.go [hunk 14]
```diff
@@ -2736,15 +2801,15 @@ func testUserStoreSearchNotInTeam(t *testing.T, ss store.Store) {
...
+	_, _ = ss.Team().SaveMember(&model.TeamMember{TeamId: teamId1, UserId: u6.Id}, -1)
 
 	teamId2 := model.NewId()
-	store.Must(ss.Team().SaveMember(&model.TeamMember{TeamId: teamId2, UserId: u4.Id}, -1))
+	_, _ = ss.Team().SaveMember(&model.TeamMember{TeamId: teamId2, UserId: u4.Id}, -1)
```
Comment (jespino) [inline store/storetest/user_store.go]: I miss the `require.Nil(t, err)` here

File: store/storetest/user_store.go [hunk 15]
```diff
@@ -2887,7 +2952,7 @@ func testUserStoreSearchWithoutTeam(t *testing.T, ss store.Store) {
 	defer func() { store.Must(ss.Bot().PermanentDelete(u3.Id)) }()
 
 	tid := model.NewId()
-	store.Must(ss.Team().SaveMember(&model.TeamMember{TeamId: tid, UserId: u3.Id}, -1))
+	_, _ = ss.Team().SaveMember(&model.TeamMember{TeamId: tid, UserId: u3.Id}, -1)
```
Comment (jespino) [inline store/storetest/user_store.go]: I miss the `require.Nil(t, err)` here

File: app/auto_users.go (hunk 1 above)
Comment (Inconnu08) [reply app/auto_users.go]: There was a merge conflict so these new changes were merged.

Comment (Inconnu08) [reply app/auto_users.go]: I'm fixing this

File: store/sqlstore/team_store.go (hunk 2 above)
Comment (Inconnu08) [reply store/sqlstore/team_store.go]: sure

File: store/storetest/team_store.go (hunk 6 above)
Comment (Inconnu08) [reply store/storetest/team_store.go]: alright

File: store/storetest/user_store.go (hunk 8 above)
Comment (Inconnu08) [reply store/storetest/user_store.go]: oops

**Comment by Inconnu08** [comment]
@jespino updated

File: store/storetest/user_store.go [hunk 16]
```diff
@@ -2736,15 +2812,19 @@ func testUserStoreSearchNotInTeam(t *testing.T, ss store.Store) {
...
-
+	_, err := ss.Team().SaveMember(&model.TeamMember{TeamId: teamId1, UserId: u1.Id}, -1)
+	require.Nil(t, err)
+	_, err = ss.Team().SaveMember(&model.TeamMember{TeamId: teamId1, UserId: u2.Id}, -1)
+	require.Nil(t, err)
+	_, err = ss.Team().SaveMember(&model.TeamMember{TeamId: teamId1, UserId: u3.Id}, -1)
+	require.Nil(t, err)
+	_, err = ss.Team().SaveMember(&model.TeamMember{TeamId: teamId1, UserId: u5.Id}, -1)
+	require.Nil(t, err)
+	_, err = ss.Team().SaveMember(&model.TeamMember{TeamId: teamId1, UserId: u6.Id}, -1)
```
Comment (jespino) [inline store/storetest/user_store.go]: missing 1 `require.Nil(t, err)` here

File: manualtesting/manual_testing.go [hunk 17]
```diff
@@ -100,7 +100,9 @@ func manualTest(c *web.Context, w http.ResponseWriter, r *http.Request) {
 		}
 
 		c.App.Srv.Store.User().VerifyEmail(user.Id, user.Email)
-		<-c.App.Srv.Store.Team().SaveMember(&model.TeamMember{TeamId: teamID, UserId: user.Id}, *c.App.Config().TeamSettings.MaxUsersPerTeam)
+		if _, err := c.App.Srv.Store.Team().SaveMember(&model.TeamMember{TeamId: teamID, UserId: user.Id}, *c.App.Config().TeamSettings.MaxUsersPerTeam); err != nil {
```
Comment (jespino) [inline manualtesting/manual_testing.go]: You are changing here the behavior, I think we should keep it as was before just removing the `<-`.

Comment (Inconnu08) [reply manualtesting/manual_testing.go]: So just ignore the output with `_`?

Comment (jespino) [reply manualtesting/manual_testing.go]: yes and no, my idea is to ignore the output but just not assigning the output to anything. a line with just `c.App.Srv.Store.Team().SaveMember(&model.TeamMember{TeamId: teamID, UserId: user.Id}, *c.App.Config().TeamSettings.MaxUsersPerTeam)`

Comment (Inconnu08) [reply manualtesting/manual_testing.go]: alright

**Comment by jespino** [comment]
@Inconnu08 Do you mind to resolve the conflicts?

**Comment by Inconnu08** [comment]
@jespino done

File: app/auto_users.go [hunk 18]
```diff
@@ -48,12 +47,17 @@ func (a *App) CreateBasicUser(client *model.Client4) *model.AppError {
 		if resp.Error != nil {
 			return resp.Error
 		}
-		_, err := a.Srv.Store.User().VerifyEmail(ruser.Id, ruser.Email)
+
+		_, err := a.Srv.Store.Team().SaveMember(&model.TeamMember{TeamId: basicteam.Id, UserId: ruser.Id}, *a.Config().TeamSettings.MaxUsersPerTeam)
```
Comment (jespino) [inline app/auto_users.go]: You changed the order of the execution of the methods here.

File: store/store.go [hunk 19]
```diff
@@ -100,13 +100,13 @@ type TeamStore interface {
...
 	AnalyticsTeamCount() (int64, *model.AppError)
-	SaveMember(member *model.TeamMember, maxUsersPerTeam int) StoreChannel
-	UpdateMember(member *model.TeamMember) (*model.TeamMember, *model.AppError)
-	GetMember(teamId string, userId string) (*model.TeamMember, *model.AppError)
-	GetMembers(teamId string, offset int, limit int, restrictions *model.ViewUsersRestrictions) ([]*model.TeamMember, *model.AppError)
-	GetMembersByIds(teamId string, userIds []string, restrictions *model.ViewUsersRestrictions) ([]*model.TeamMember, *model.AppError)
-	GetTotalMemberCount(teamId string) (int64, *model.AppError)
-	GetActiveMemberCount(teamId string) (int64, *model.AppError)
+	SaveMember(member *model.TeamMember, maxUsersPerTeam int) (*model.TeamMember, *model.AppError)
+	UpdateMember(member *model.TeamMember) StoreChannel
```
Comment (jespino) [inline store/store.go]: You merged it wrongly. You remove all the changes from other PRs and use yours. You should pick your changes and apply it to the version from master.

File: store/storetest/user_store.go [hunk 20]
```diff
@@ -301,24 +309,27 @@ func testGetAllUsingAuthService(t *testing.T, ss store.Store) {
...
 		Email:       MakeEmail(),
 		Username:    "u3" + model.NewId(),
 		AuthService: "service2",
 	})).(*model.User)
 	defer func() { require.Nil(t, ss.User().PermanentDelete(u3.Id)) }()
-	store.Must(ss.Team().SaveMember(&model.TeamMember{TeamId: teamId, UserId: u3.Id}, -1))
-	_, err := ss.Bot().Save(&model.Bot{
+	_, err = ss.Team().SaveMember(&model.TeamMember{TeamId: teamId, UserId: u3.Id}, -1)
+	require.Nil(t, err)
+	store.Must(ss.Bot().Save(&model.Bot{
```
Comment (jespino) [inline store/storetest/user_store.go]: You merged it wrong, you should respect the master changes, Bot.Save was already migrated, so you should maintain the master approach (sync by default) for that methods, so... `store.Must` doesn't makes sense.

File: store/storetest/user_store.go [hunk 21]
```diff
@@ -662,22 +679,25 @@ func testUserStoreGetProfilesInChannel(t *testing.T, ss store.Store) {
...
 	u3 := store.Must(ss.User().Save(&model.User{
 		Email:    MakeEmail(),
 		Username: "u3" + model.NewId(),
 	})).(*model.User)
 	defer func() { require.Nil(t, ss.User().PermanentDelete(u3.Id)) }()
-	store.Must(ss.Team().SaveMember(&model.TeamMember{TeamId: teamId, UserId: u3.Id}, -1))
-	_, err := ss.Bot().Save(&model.Bot{
+	_, err = ss.Team().SaveMember(&model.TeamMember{TeamId: teamId, UserId: u3.Id}, -1)
+	require.Nil(t, err)
+	store.Must(ss.Bot().Save(&model.Bot{
```
Comment (jespino) [inline store/storetest/user_store.go]: Same problem here

File: store/storetest/user_store.go [hunk 22]
```diff
@@ -755,22 +775,25 @@ func testUserStoreGetProfilesInChannelByStatus(t *testing.T, ss store.Store) {
...
```
Comment (jespino) [inline store/storetest/user_store.go]: Same problem here

File: store/storetest/user_store.go [hunk 23]
```diff
@@ -904,22 +928,25 @@ func testUserStoreGetAllProfilesInChannel(t *testing.T, ss store.Store) {
...
```
Comment (jespino) [inline store/storetest/user_store.go]: Same problem here.

File: store/storetest/user_store.go [hunk 24]
```diff
@@ -1016,22 +1043,25 @@ func testUserStoreGetProfilesNotInChannel(t *testing.T, ss store.Store) {
...
```
Comment (jespino) [inline store/storetest/user_store.go]: Same problem here

File: store/storetest/user_store.go [hunk 25]
```diff
@@ -1162,22 +1192,25 @@ func testUserStoreGetProfilesByIds(t *testing.T, ss store.Store) {
...
```
Comment (jespino) [inline store/storetest/user_store.go]: Same problem here

File: store/storetest/user_store.go [hunk 26]
```diff
@@ -1226,22 +1259,25 @@ func testUserStoreGetProfilesByUsernames(t *testing.T, ss store.Store) {
...
 	u3 := store.Must(ss.User().Save(&model.User{
 		Email:    MakeEmail(),
 		Username: "u3" + model.NewId(),
 	})).(*model.User)
 	defer func() { require.Nil(t, ss.User().PermanentDelete(u3.Id)) }()
-	store.Must(ss.Team().SaveMember(&model.TeamMember{TeamId: team2Id, UserId: u3.Id}, -1))
-	_, err := ss.Bot().Save(&model.Bot{
+	_, err = ss.Team().SaveMember(&model.TeamMember{TeamId: team2Id, UserId: u3.Id}, -1)
+	require.Nil(t, err)
+	store.Must(ss.Bot().Save(&model.Bot{
```
Comment (jespino) [inline store/storetest/user_store.go]: Same problem here

File: store/storetest/user_store.go [hunk 27]
```diff
@@ -1290,23 +1326,26 @@ func testUserStoreGetSystemAdminProfiles(t *testing.T, ss store.Store) {
...
 		Email:    MakeEmail(),
 		Roles:    model.SYSTEM_USER_ROLE_ID + " " + model.SYSTEM_ADMIN_ROLE_ID,
 		Username: "u3" + model.NewId(),
 	})).(*model.User)
 	defer func() { require.Nil(t, ss.User().PermanentDelete(u3.Id)) }()
-	store.Must(ss.Team().SaveMember(&model.TeamMember{TeamId: teamId, UserId: u3.Id}, -1))
-	_, err := ss.Bot().Save(&model.Bot{
+	_, err = ss.Team().SaveMember(&model.TeamMember{TeamId: teamId, UserId: u3.Id}, -1)
+	require.Nil(t, err)
+	store.Must(ss.Bot().Save(&model.Bot{
```
Comment (jespino) [inline store/storetest/user_store.go]: Same problem here and more in the rest of the file

File: app/auto_users.go [hunk 28]
```diff
@@ -48,12 +47,18 @@ func (a *App) CreateBasicUser(client *model.Client4) *model.AppError {
...
 			return resp.Error
 		}
+
 		_, err := a.Srv.Store.User().VerifyEmail(ruser.Id, ruser.Email)
 		if err != nil {
 			return err
 		}
-		store.Must(a.Srv.Store.Team().SaveMember(&model.TeamMember{TeamId: basicteam.Id, UserId: ruser.Id}, *a.Config().TeamSettings.MaxUsersPerTeam))
+
+		_, err = a.Srv.Store.Team().SaveMember(&model.TeamMember{TeamId: basicteam.Id, UserId: ruser.Id}, *a.Config().TeamSettings.MaxUsersPerTeam)
```
Comment (jespino) [inline app/auto_users.go]: This is not exactly the same previous behavior, but I think it can be valid anyway.

File: store/store.go [hunk 29]
```diff
@@ -100,13 +100,13 @@ type TeamStore interface {
 	GetByInviteId(inviteId string) (*model.Team, *model.AppError)
 	PermanentDelete(teamId string) StoreChannel
 	AnalyticsTeamCount() (int64, *model.AppError)
-	SaveMember(member *model.TeamMember, maxUsersPerTeam int) StoreChannel
+	SaveMember(member *model.TeamMember, maxUsersPerTeam int) (*model.TeamMember, *model.AppError)
 	UpdateMember(member *model.TeamMember) (*model.TeamMember, *model.AppError)
-	GetMember(teamId string, userId string) (*model.TeamMember, *model.AppError)
```
Comment (jespino) [inline store/store.go]: You are still "undoing" the migrations made by others, in this diff you should only see your changes, not changes related to other functions.

File: store/storetest/user_store.go [hunk 30]
```diff
@@ -1016,29 +1043,33 @@ func testUserStoreGetProfilesNotInChannel(t *testing.T, ss store.Store) {
...
+	require.Nil(t, err)
+	_, err = ss.Bot().Save(&model.Bot{
 		UserId:   u3.Id,
 		Username: u3.Username,
 		OwnerId:  u1.Id,
 	})
 	require.Nil(t, err)
 	u3.IsBot = true
 	defer func() { require.Nil(t, ss.Bot().PermanentDelete(u3.Id)) }()
+	defer func() { require.Nil(t, ss.User().PermanentDelete(u3.Id)) }()
```
Comment (jespino) [inline store/storetest/user_store.go]: You are adding here a line that wasn't there before, and I think can have more implications than I expect. Just leave it as was before.

File: app/auto_users.go (hunk 28 above)
Comment (Inconnu08) [reply app/auto_users.go]: Alright, not changing this then @jespino

**Comment by Inconnu08** [comment]
closing as its way outdated
